### PR TITLE
p-59 runtime upgrade simulation for litmus

### PR DESCRIPTION
**Pull request by 0xverin**
resolve #p-59
this pr submitted a commit in https://github.com/litentry/fork-off-substrate/commit/a80ce017fb9c67bf45089b760d5dbdf29504e2c7

### CODE REVIEW

**Comment by Kailai-Wang** [comment]
Do you have a test run of the simulaiton implementation? If so could you attach the link?

**Comment by 0xverin** [comment]
Updates:
[notePreimage](https://github.com/litentry/litentry-parachain/actions/runs/8799785778/job/24149541269#step:6:48) and [proposal](https://github.com/litentry/litentry-parachain/actions/runs/8799785778/job/24149541269#step:6:54) have both submitted successfully.

but,currently, the test is stuck on the limitation of launchPeriod.
In CI, we did indeed use the [fast-runtime feature](https://github.com/litentry/litentry-parachain/blob/dev/.github/workflows/ci.yml#L413).
But through polkadot.js queries, the launchPeriod is still limited to [36000 blocks](https://github.com/litentry/litentry-parachain/actions/runs/8799785778/job/24149541269#step:6:40) (5 days).
So waiting [here](https://github.com/litentry/litentry-parachain/pull/2679/files#diff-a1d2895ee4b98d71c62327c89359427ff8765fc7ea354e3a5744cfff43e67092R25) for 6 minutes still won't wait for the democracy event, which will block voted.
Does anyone have any ideas about this? How to make the launchPeriod of litmus 5 minutes? cc @litentry/parachain 

image

File: docker/fork.json
Comment (Kailai-Wang) [inline docker/fork.json:1]: I don't think we need this file committed

File: ts-tests/integration-tests/runtime-upgrade.test.ts [hunk 1]
```diff
@@ -52,18 +88,134 @@ async function runtimeUpgrade(api: ApiPromise, wasm: string) {
...
+    const network = process.argv[7].slice(2);
+    console.log('Running runtime upgrade test on network:', network);
     step('Running runtime ugprade test', async function () {
+        let runtimeVersion: number;
         const wasmPath = path.resolve('/tmp/runtime.wasm');
+        console.log(`wasmPath: ${wasmPath}`);
         const wasm = fs.readFileSync(wasmPath).toString('hex');
-        const runtimeVersion = await runtimeUpgrade(context.api, `0x${wasm}`);
+        if (network === 'rococo' || network === 'litentry') {
+            runtimeVersion = await runtimeUpgradeWithSudo(context.api, `0x${wasm}`);
```
Comment (Kailai-Wang) [inline ts-tests/integration-tests/runtime-upgrade.test.ts]: Litentry doesn't have sudo either

**Review by Kailai-Wang — COMMENTED** [review]
Indeed - if you use release wasm (where no fast-runtime is applied when building), the launch period will be configured as prod

Have you checked how other projects tackle this?

**Comment by 0xverin** [comment]
updates:
Simulated upgrade test completed, please go to the [CI results](https://github.com/litentry/litentry-parachain/actions/runs/9145349048/job/25144192922#step:6:98). Please ignore litmus error because I used a lower version(9176) of wasm to upgrade the litmus runtime, but the runtime upgrade has already been applied. litentry and rococo both passed

Before testing, need to merge this fork-off-substrate [pr](https://github.com/litentry/fork-off-substrate/pull/5). cc @Kailai-Wang @litentry/parachain

**Review by Kailai-Wang — APPROVED** [review]
Thanks - I'm happy as long as the test run looks good!

The PR in fork-off-substrate is merged.

Let's see how it behaves in next release : )

**Pull request merged**